endpoints.tf

```diff
@@ -1,3 +1,7 @@
+locals {
+  s3_route_table_ids = var.s3_endpoint.route_table_ids != null ? var.s3_endpoint.route_table_ids : aws_route_table.private[*].id
+}
```

Comment:
looks like you need to drop the .route_table_ids, like:
```
s3_route_table_ids = var.s3_endpoint != null ? var.s3_endpoint.route_table_ids : aws_route_table.private[*].id
```

Reply:
good catch